packages/core/src/codewhisperer/service/transformationHubViewProvider.ts

```diff
@@ -120,15 +120,20 @@ export class TransformationHubViewProvider implements vscode.WebviewViewProvider
             planSteps = await getTransformationSteps(transformByQState.getJobId())
         }
         let progressHtml = `<p><b>Transformation Status</b></p><p>No job ongoing</p>`
```

Comment:
Question: If its in scope for your timeline, it would be a "nice to have" to refactor the `showPlanProgress` to use string literals to build the progress panel markup.

This would be a significant refactor of this function, into several smaller functions, but would help readability and extensibility in the future.

See example refactor:
```
    
    private renderTransformationHeader() {
        return "<p><b>Transformation Status</b></p><p>No job ongoing</p>"

    }

    private renderStartJobSucceededStep(planProgress) {
        const startJobSucceededFlag = planProgress['startJob'] === StepProgress.Succeeded
        if (!startJobSucceededFlag) return ''

        return `<p> 
            ${this.getProgressIconMarkup(planProgress['buildCode'])} 
            Build uploaded code in secure build environment
        </p>
        `
    }

    private renderBuildJobSucceededStep(planProgress) {
        const buildCodeSucceededFlag = planProgress['buildCode'] === StepProgress.Succeeded
        if (!buildCodeSucceededFlag) return ''

        return `<p> 
            ${this.getProgressIconMarkup(planProgress['generatePlan'])} 
            Generate transformation plan
        </p>
        `
    }

    private renderPlanStepsSucceededStep(planProgress, planSteps) {
        const generatePlanSucceededFlag = planProgress['generatePlan'] === StepProgress.Succeeded
        if (!generatePlanSucceededFlag) return ''

        return `<p> 
            ${this.getProgressIconMarkup(planProgress['transformCode'])} 
            Transform your code to Java 17 using transformation plan
        </p>
        `
            // TODO Convert/refactor below if statement into string literal
            if () {
                progressHtml += `<p> 
            ${this.getProgressIconMarkup(planProgress['transformCode'])} 
            Transform your code to Java 17 using transformation plan
        </p>`
                // now get the details of each sub-step of the "transformCode" step
                if (planSteps !== undefined) {
                    const stepStatuses = []
                    for (const step of planSteps) {
                        stepStatuses.push(step.status)
                    }
                    for (let i = 0; i < planSteps.length; i++) {
                        const step = planSteps[i]
                        const stepStatus = step.status
                        let stepProgress = undefined
                        if (stepStatus === 'COMPLETED' || stepStatus === 'PARTIALLY_COMPLETED') {
                            stepProgress = StepProgress.Succeeded
                        } else if (
                            stepStatus === 'STOPPED' ||
                            stepStatus === 'FAILED' ||
                            planProgress['transformCode'] === StepProgress.Failed
                        ) {
                            stepProgress = StepProgress.Failed
                        } else {
                            stepProgress = StepProgress.Pending
                        }
                        // include check for the previous step not being CREATED, as this means it has finished, so we can display the next step
                        if (step.startTime && step.endTime && (i === 0 || stepStatuses[i - 1] !== 'CREATED')) {
                            const stepTime = step.endTime.toLocaleDateString('en-US', {
                                hour: '2-digit',
                                minute: '2-digit',
                                second: '2-digit',
                            })
                            const stepDuration = convertToTimeString(step.endTime.getTime() - step.startTime.getTime())
                            progressHtml += `<p style="margin-left: 20px">${this.getProgressIconMarkup(stepProgress)} ${
                                step.name
                            } [finished on ${stepTime}] <span style="color:grey">${stepDuration}</span></p>`
                        } else if (stepStatuses[i - 1] !== 'CREATED') {
                            progressHtml += `<p style="margin-left: 20px">${this.getProgressIconMarkup(stepProgress)} ${
                                step.name
                            }</p>`
                        }
                        if (step.progressUpdates) {
                            for (const subStep of step.progressUpdates) {
                                progressHtml += `<p style="margin-left: 40px">- ${subStep.name}</p>`
                                if (subStep.description) {
                                    progressHtml += `<p style="margin-left: 60px; color:${this.getFontColorForSubStep(
                                        subStep.status
                                    )}">- ${subStep.description}</p>`
                                }
                            }
                        }
                    }
                }
            }
    }

    private renderProgressStepDetails(planProgress, planSteps) {

        return `
            ${renderStartJobSucceededStep(planProgress)}
            ${renderBuildJobSucceededStep(planProgress)}
            ${renderPlanStepsSucceededStep(planProgress, planSteps)}
        `
    }

    private renderTransformationStepBody(planProgress, planSteps) {
        if (planProgress['transformCode'] === StepProgress.NotStarted) return ''

        return `
            <section>
                <p><b>Transformation Status</b></p>
                <p> ${this.getProgressIconMarkup(planProgress['startJob'])} Waiting for job to start</p>
                ${renderProgressStepDetails(planProgress, planSteps)}
            </section>
        `
    }

    private renderProgressHtml(planProgress, planSteps) {
        return `
            ${renderTransformationHeader()}
            ${renderTransformationStepBody(planProgress, planSteps)}
        `
    }
    
    private async showPlanProgress(startTime: number): Promise<string> {
        const planProgress = getPlanProgress()
        let planSteps = undefined
        if (planProgress['buildCode'] === StepProgress.Succeeded) {
            planSteps = await getTransformationSteps(transformByQState.getJobId())
        }
        const progressHtml = renderProgressHtml(planProgress, planSteps)
    }
```

Reply:
I see some uses like this in the VSCode repository:
```
        if (vscode.window.activeColorTheme.kind === vscode.ColorThemeKind.Light) {
            panel.iconPath = vscode.Uri.file(
                context.asAbsolutePath(join('resources', 'icons', 'aws', 'applicationcomposer', 'icon.svg'))
            )
        } else {
            panel.iconPath = vscode.Uri.file(
                context.asAbsolutePath(join('resources', 'icons', 'aws', 'applicationcomposer', 'icon-dark.svg'))
            )
        }
        ```

Reply:
For IntelliJ you can check the `CodeModernizerUIConstants.kt` [file](https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/constants/CodeModernizerUIConstants.kt) and see we have constants to set a consistent, IDE specific RED/Green we use throughout our UI:

```
        fun getGreenThemeFontColor(): Color = if (JBColor.isBright()) JBColor.GREEN.darker() else JBColor.GREEN
        fun getRedThemeFontColor(): Color = JBColor.RED
        ```

Reply:
We also have the mynah-ui CSS variables available -> https://github.com/aws/mynah-ui/blob/5d128f1e4f0e7931cfc8ff6c774bd770d0a49a81/docs/STYLING.md

I'm torn here because we are integrating with their product/design system and should look seamless with them. On the other hand we should also look seamless with the IDE and if their brand colors are different, we will look different.

Reply:
I can access the color theme kind (options are Light, High Contrast Light, Dark, High Contrast Dark) but the problem is which exact shade of red/green do I use for each kind? The mynah-ui CSS doesn't define exact shades of each color to use; it lets you customize the shade with a hex code. So yes we can do this but I need the specific color we want for each circumstance. Elsewhere in the code (for the green/grey checkmarks), I just use standard green/gray, so currently the implementation at least matches that.

Reply:
I think for IntelliJ we just covered:
- light themes get a darker red/green
- dark themes get a light red/green

We did not cover individual themes, because there are too many and custom ones. We went with a "good enough" approach covering the base dark/light themes. I think for VSCode we can do the same.